zstacklib/zstacklib/utils/file_system_upload_task.py

```
@@ -0,0 +1,85 @@
+logger = log.get_logger(__name__)
+
+class FileObject(StorageObject):
+    def __init__(self, file_path):
+        super(FileObject, self).__init__()
+        self.file_path = file_path
+        self.offset = 0
+        try:
+            self.file_obj = open(file_path, 'r+b')
+            self.size = os.path.getsize(file_path)
+        except (IOError, OSError) as e:
+            raise type(e)("Failed to open file %s: %s" % (file_path, str(e)))
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)